src/java/org/apache/cassandra/index/sai/StorageAttachedIndex.java

```diff
@@ -390,10 +390,15 @@ public Callable<?> getPreJoinTask(boolean hadBootstrap)
     public Callable<?> getTruncateTask(long truncatedAt)
     {
         /*
-         * index files will be removed as part of base sstable lifecycle in
-         * {@link LogTransaction#delete(java.io.File)} asynchronously.
+         * index files will be removed as part of base sstable lifecycle in {@link LogTransaction#delete(java.io.File)}
+         * asynchronously, but we need to mark the index queryable because if the truncation is during the initial
+         * build of the index it won't get marked queryable by the build.
```

Comment:
If the index is currently marked as non-queryable, and we truncate, is this safe because truncation will remove all the on-disk artifacts (or lack of those artifacts) that made the index non-queryable in the first place?

Reply:
I am working on the basis that if we truncate we will delete any existing data (sstable and index). As a result, we can safely mark the index as queryable and let further errors mark it not-queryable.

Reply:
Most clusters I work on don't even allow truncation in production :p